File: AntOnACube/Ant.cpp
#include "Ant.h"

Ant::Ant(Vertex* start, Vertex* goal, Graph* graph) {
	this->location = start;
	this->goal = goal;
	this->graph = graph;
	totalTimeTraveled = 1;
	path.push_back(start);
}

void Ant::move(RandomNumberGenerator* rng) {
	int pathChoice;
	rng->getRandomNumber(&pathChoice);
	location = (graph->adjacencyMatrix[location])->at(pathChoice);
	path.push_back(location);
	this->totalTimeTraveled++;
	if (location != goal) {
		move(rng);
	}
}

File: AntOnACube-py/Ant.py
import random
from collections import defaultdict

class Ant(object):
    path = []
    weight = 0
    graph = None
    location = None
    goal = None

    def __init__(self, graph: defaultdict, location: int, goal: int):
        self.graph = graph
        self.location = location
        self.goal = goal
        self.path.append(self.location)

    def move(self):
        choice = random.randint(0,2)
        self.location = self.graph[self.location][choice]
        self.path.append(self.location)
        self.weight += 1
        if self.location != self.goal:
            self.move()





